posthog/queries/test/test_trend_combinations.py

```diff
@@ -0,0 +1,39 @@
+from posthog.models import Event, Person
+from posthog.models.filters.filter import Filter
+from posthog.models.team import Team
+from posthog.queries.test.trends import all_trend_tests
+from posthog.queries.test.trends.base import QueryTest
+from posthog.queries.trends import Trends
+from posthog.test.base import TransactionBaseTest
+
+
+# TODO: add more supported entities here
+def setup_test(team: Team, payload):
+    for entity, data in payload.items():
+        if entity == "events":
+            for event_data in data:
+                Event.objects.create(team=team, **event_data)
+        if entity == "people":
+            for person_data in data:
+                Person.objects.create(team=team, **person_data)
+
+
+# need to generate tests without using subtest so that the transactions will be cleared after each test
```

Comment:
REF: https://stackoverflow.com/questions/32899/how-do-you-generate-dynamic-parameterized-unit-tests-in-python